sailor/sap_iot/fetch_aggregates.py

```diff
@@ -0,0 +1,195 @@
+"""
+Timeseries module can be used to retrieve timeseries data from the SAP iot abstract timeseries api.
+
+This is the equivalent to sap_iot.fetch, except we're retrieving pre-aggregated data from the hot store rather than
+raw data from the cold store.
+"""
+
+from __future__ import annotations
+
+from typing import Union, Iterable, TYPE_CHECKING
+from datetime import datetime, timedelta
+import warnings
+import re
+import logging
+from collections import defaultdict
+
+import pandas as pd
+
+from sailor.utils.timestamps import _any_to_timestamp, _timestamp_to_isoformat
+from sailor.assetcentral.indicators import AggregatedIndicatorSet, IndicatorSet
+from sailor.sap_iot.wrappers import TimeseriesDataset
+from sailor.utils.oauth_wrapper import get_oauth_client
+from sailor.sap_iot._common import request_aggregates_url
+
+LOG = logging.getLogger(__name__)
+LOG.addHandler(logging.NullHandler())
+
+if TYPE_CHECKING:
+    from ..assetcentral.equipment import EquipmentSet
+
+
+_ALL_KNOWN_AGGREGATION_FUNCTIONS = ['MIN', 'AVG', 'FIRST', 'MAX', 'SUM', 'LAST', 'COUNT',
+                                    'PERCENT_GOOD', 'TMAX', 'TLAST', 'STDDEV', 'TFIRST', 'TMIN']
+
+
+def _parse_aggregation_interval(aggregation_interval):
+    if aggregation_interval is not None and not isinstance(aggregation_interval, str):
+        total_seconds = aggregation_interval.total_seconds()
+        aggregation_interval = f'PT{total_seconds}S'
+    return aggregation_interval
```

Comment:
do we actually want to allow setting the aggregation interval to "None"? 
what you get back is a single data point aggregating the whole duration. this behavior is also not clearly defined, is it? it just happens.
do you see any benefits of offering that?

Reply:
as discussed, let's leave it possible to use None